app/decorators/odk/form_decorator.rb

```diff
@@ -45,5 +45,17 @@ def ir_code_binding
         type: "string"
       )
     end
+
+    # Whether this form needs an accompanying manifest for odk.
+    def needs_manifest?
+      # For now this is IFF there are any multilevel option sets or questions with audio prompts
+      @needs_manifest ||= option_sets.any?(&:multilevel?) || audio_prompt_content?
+    end
+
+    private
+
+    def audio_prompt_content?
+      smsable_questionings.values.any?(&:audio_prompt_content_type)
```

Comment:
Good attempt! 

I'm curious why you used `smsable_questionings` here. I'm guessing b/c it turns out there is no `visible_questionings` method and this was the closest thing? In cases like this you should please ask me rather than guess. `smsable_questionings` won't do here because it would exclude questions we don't want to exclude.

Check out the `questions` method in `Form`. Then you can also get rid of the delegator.

Also, there is a cleaner way to test if a Paperclip attachment is present:

https://makandracards.com/makandra/1419-check-whether-a-paperclip-attachment-exists

Using`audio_prompt_content_type` is not a bad idea but it's purpose is not as understandable as the above.